Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -580,14 +591,13 @@ private func walkPkgTreeAndGetModuleAliases(for pkgID: PackageIdentity,
                     break
                 }
             }
-            
-            // Add pkgID to a stack used to keep track of multiple
-            // aliases specified in the package chain. Need to add
-            // pkgID here, otherwise need pkgID != pkgInChain check
-            // in the for loop above
-            pkgStack.append(pkgID)
         }
-        
+        // Add pkgID to a stack used to keep track of multiple
+        // aliases specified in the package chain. Need to add
+        // pkgID here, otherwise need pkgID != pkgInChain check
+        // in the for loop above
+        pkgStack.append(pkgID)
```

Comment:
this change seems unrelated but potentially material?

Reply:
This change makes sure pkgID is only added once, so it's an improvement.